### ISSUE TITLE
[BUG]LinAlgError: Matrix is not positive definite

### ISSUE THREAD

**Issue by NamrataRade** [sktime/sktime#4265]
**Describe the bug**

I am using VARIMA model from sktime package for time series forecasting. Input data has product informations at different levels. Goal is to do hirarchical time series forecasting. Before passing input data, I am performing log transformation and stationary check. inut data size=289065.After running model.fit(train_ts) command getting this LinAlgError. For small sample e.g 325 data points, I am not getting this error.

Input data have two columns as an index: unique_id and time. Other columns are:sales_value and area_id. 
-->

```python
<```
horizon="2022-02-28"
y_val = 'sales_value'
time_col= 'time'
hier_columns = ['unique_id','area_id','lvl1','lvl2','lvl3','lvl4','lvl5','lvl6','lvl7','lvl8','lvl9']
hier_df = df_monthly_new[hier_columns + [y_val] + [time_col]]
df_final=hier_df.copy()
df_final.shape
train_df = df_final[df_final['time']<horizon]
valid_df = df_final[df_final['time']>=horizon]
from sktime.forecasting.varmax import VARMAX
forecaster = VARMAX(suppress_warnings=True,enforce_stationarity=False)  
forecaster.fit(train_df[['sales_value','area_id']])  >
```
**Expected behavior**

**Additional context**

**Versions**
<details>

</System:
    python: 3.10.9 (main, Jan 11 2023, 15:21:40) [GCC 11.2.0]
executable: /anaconda/envs/azureml_py310_sdkv2/bin/python
   machine: Linux-5.15.0-1031-azure-x86_64-with-glibc2.31

Python dependencies:
          pip: 22.3.1
   setuptools: 65.6.3
      sklearn: 1.2.0
       sktime: 0.16.1
  statsmodels: 0.13.5
        numpy: 1.23.5
        scipy: 1.10.0
       pandas: 1.5.2
   matplotlib: 3.6.3
       joblib: 1.2.0
        numba: 0.56.4
     pmdarima: 2.0.2
      tsfresh: None>

**Comment by NamrataRade** [comment]
Input columns       
---  ------       --------------   -----         
 0   lvl1         289065 non-null  category      
 1   lvl2         289065 non-null  category      
 2   lvl3         289065 non-null  category      
 3   lvl4         289065 non-null  category      
 4   lvl5         289065 non-null  category      
 5   lvl6         289065 non-null  category      
 6   lvl7         289065 non-null  category      
 7   lvl8         289065 non-null  category      
 8   lvl9         289065 non-null  category      
 9   area_id      289065 non-null  int16         
 10  time         289065 non-null  datetime64[ns]
 11  sales_value  289065 non-null  float16       
 12  unique_id    289065 non-null  int32         
dtypes: category(9), datetime64[ns](1), float16(1), int16(1), int32(1)
memory usage: 7.5 MB

**Comment by fkiraly** [comment]
Thanks for reporting!

`VARMAX` is a direct interface to `statsmodels` (giving it the `sktime` unified interface), and this sounds like it might be coming from the underlying model.

To diagnose, may I ask:

can you try fitting `VARMAX` from `statsmodels.tsa.statespace.varmax` on the same data, and see whether it crashes?
If yes, I would recommend to go "one layer down" and open an issue in `statsmodels` repo (and kindly link it here).

Also, I notice we have interfaces to ARIMA (univariate auto-regressive) models by different implementers (statsmodels, pmdarima, nixitla/statsforecast), but not for VARMA (or vector auto-regressive more generally) - if you know alternatives and can point to them, it should be quick to build an interface.

**Comment by NamrataRade** [comment]
Thank you for the reply
I have tried VAR and VARMAX model from statsmodels. It is working fine and able to train on the entire dataset.

**Comment by fkiraly** [comment]
> I have tried VAR and VARMAX model from statsmodels. It is working fine and able to train on the entire dataset.

That's odd - have you tried the `sktime` interface multiple times? Is the error reproducible, or sporadic?

**Comment by NamrataRade** [comment]
For different blocks(samples) of data, it is working but not for the entire data together.

**Comment by fkiraly** [comment]
ok, so to summarize, please correct me if I'm wrong:

* you run `sktime` `VARMAX` on a certain dataset, it *always* errors out with `LinAlgError` (if you repeat multiple times)
* you run `statsmodels` VARMAX` on precisely the same data, it never errors out (if you repeat multiple times)

do I understand this correctly?

**Comment by NamrataRade** [comment]
In the input data, there are 200000 data points. Sktime's VARMAX model gives result only for sample data e.g 250 data points. Once we get result for first 250 sample, we can repeat the process for another 250 samples. But if I pass around 180000 data points as a input to model, it gives this error. But for statsmodels VARMAX, model is able to train on 180000 data in a single iteration. Can you guide me on what kind of input data should I pass to model. For now, I have tried by giving wide format(time column as an index and pivoted dataframe with multiple sales value column which belongs to different products combinations) and long format(where time and level column as index and sales value and unique I'd column as y).

**Comment by fkiraly** [comment]
Wait a sec, there's something I don't understand at the moment: how do you pass the hierarchical data to `statsmodels` `VARMAX`? As far as I know, it does not have a native hierarchical mode; whereas `sktime` iterates over lowest level nodes and fits multiple `statsmodels` `VARMAX` models.

Could you share your `statsmodels` based code for comparison?

**Comment by NamrataRade** [comment]
I passed dataframe with three columns only. Time column as index and unique_id(which represents level information) and sales value as y. This is not the right data to pass. The idea was to use automatic packages models for this projects. We also implemented other models like arima, exponential smoothening,theta etc using sktime. Those are running fine. Only VAR model is giving error. so I thought to check with statsmodels. I have tried by giving one more input data format which time column as index and sales value and sales value shifted by one(sales value lag shift1). With this combination, sktime is not giving an error even if I am passing entire data to train. But if I am increasing p value of model to more than 1, again it is throwing same error.

**Comment by NamrataRade** [comment]
model= VARMAX(train_data).fit(disp=False)
Y_pred=model.predict(start='', end='')

___Statsmodels commands

**Comment by fkiraly** [comment]
ok, but this way you are not passing the data as hierarchical to `statsmodels`, so the `sktime` and `statsmodels` situations are not comparable.

What I'm now suspecting that there might be one level (unique_id) that has a very low number of data points that breaks the `VARMAX` model being fitted to it. Could that be the case?

**Comment by NamrataRade** [comment]
There are 8739 unique ids. But it is failing with different format as well. Example- pivoted dataframe where time column(month end date) is index and remaining columns are sales values of different product combination. This is kind of similar to the inbuilt dataset used for VARMAX model in the documentation of sktime.

**Comment by fkiraly** [comment]
yes, what I mean is that we need to send comparable cases to `sktime` and `statsmodels` `VARMAX` to understand whether it is coming from `sktime` or `statsmodels`.

If you can't provide self-contained code (e.g., the data is sensitive), I would kindly ask you to be precise about what data are you passing how to `VARMAX`, in both cases (`sktime` and `statsmodels`), so I can understand why the one fails and the other not.

Can you say precisely, in both cases, how many rows, columns, index levels your data frame had, and what types (columns, index)?
